controller/src/main/java/org/jboss/as/controller/PathAddress.java

```diff
@@ -150,6 +147,68 @@ public static PathAddress pathAddress(PathAddress parent, PathElement... element
         return pathAddress(list);
     }
 
+    public static PathAddress parseAddress(String address) throws IllegalArgumentException {
+        PathAddress parsedAddress = PathAddress.EMPTY_ADDRESS;
+        if (address == null || address.trim().isEmpty()) {
+            return parsedAddress;
+        }
+        String trimmedAddress = address.trim();
+        if (trimmedAddress.charAt(0) != '/' || !Character.isAlphabetic(trimmedAddress.charAt(1))) {
+            throw new IllegalArgumentException();
+        }
+        char[] characters = address.toCharArray();
+        boolean escaped = false;
+        StringBuilder keyBuffer = new StringBuilder();
+        StringBuilder valueBuffer = new StringBuilder();
+        StringBuilder currentBuffer = keyBuffer;
+        for (int i = 1; i < characters.length; i++) {
+            switch (characters[i]) {
+                case '/':
+                    if (escaped) {
+                        escaped = false;
+                        currentBuffer.append(characters[i]);
+                    } else {
+                        parsedAddress = addpathAddressElement(parsedAddress, keyBuffer, valueBuffer);
+                        keyBuffer = new StringBuilder();
+                        valueBuffer = new StringBuilder();
+                        currentBuffer = keyBuffer;
+                    }
+                    break;
+                case '\\':
+                    if (escaped) {
+                        escaped = false;
+                        currentBuffer.append(characters[i]);
+                    } else {
+                        escaped = true;
+                    }
+                    break;
+                case '=':
+                    if (escaped) {
+                        escaped = false;
+                        currentBuffer.append(characters[i]);
+                    } else {
+                        currentBuffer = valueBuffer;
+                    }
+                    break;
+                default:
+                    currentBuffer.append(characters[i]);
+                    break;
+            }
+        }
+        parsedAddress = addpathAddressElement(parsedAddress, keyBuffer, valueBuffer);
+        return parsedAddress;
+    }
+
+    private static PathAddress addpathAddressElement(PathAddress parsedAddress, StringBuilder keyBuffer, StringBuilder valueBuffer) {
+        if (keyBuffer.length() > 0) {
+            if (valueBuffer.length() > 0) {
+                return parsedAddress.append(PathElement.pathElement(keyBuffer.toString(), valueBuffer.toString()));
+            }
+            return parsedAddress.append(PathElement.pathElement(keyBuffer.toString()));
```

Comment:
I don't think we should allow this. PathElement.pathElement(String) is a convenient mechanism for server side developers, but an incompletely specified String is not valid as an end user API. If they want a wildcard, they need to use key=*.